README.md

```diff
@@ -11,3 +11,292 @@
+
+Function | Description
+-------- | -----------
+all | Make a request to the API, based on the `url` configuration setting
+first | Given a query, get the first model instance from the API
+create | Create a new instance of the model. Defaults come from the `defaults` configuration setting.
+
+> You can see full details of these methods in the [API documentation](/docs/api.md).
+
+### Custom class methods
+
+angular-model allows you to define class methods on instances. This is similar to static methods in the OOP world.
+
+```javascript
+yourApp.config(function(modelProvider) {
+    modelProvider.model('posts', {
+        // ...
+
+        /**
+         * @ngdoc object
+         * @name yourApp.posts.$class
+         * @description
+         * Class methods that are callable on the posts class, or any instance thereof. These
+         * behave similarly to static methods in OOP languages.
+         */
+        $class: {
+            /**
+             * @ngdoc function
+             * @name yourApp.posts.roles
+             * @description
+             * Get an array of valid post types.
+             *
+             * @return array The valid types that a post can have. Array of strings
+             */
+            types: function() {
+                return ['announcement', 'article']
+            }
+        }
+    });
+});
+```
+
+Example:
+
+```javascript
+console.log(model('Posts').types());
+=> ['announcement', 'article']
+```
+
+## Collection methods
+
+You can use collection methods as well, so you can deal with a bunch of instances together. This allows you to have powerful and expressive methods on collections.
+
+### Default collection methods
+
+The following methods are available statically to angular-model:
+
+Function | Description
+-------- | -----------
+add | Saves the `object` with `data`
+remove | Find `index` and delete it from the API, then remove it from the collection
+
+> You can see full details of these methods in the [API documentation](/docs/api.md).
+
+### Custom collection methods
+
+```javascript
+yourApp.config(function(modelProvider) {
+    modelProvider.model('posts', {
+        // ...
+
+        /**
+         * @ngdoc object
+         * @name yourApp.posts.$collection
+         * @description
+         * Methods that apply to a collection of posts together
+         */
+        $collection: {
+            /**
+             * @ngdoc function
+             * @name yourApp.posts.$hasArchived
+             * @description
+             * Operates on a collection of posts and determines whether any of them are archived
+             *
+             * @requires _ Lodash library is used to search the collection
+             *
+             * @return string Either the logo for this post, or a default logo
+             */
+            $hasArchived: function() {
+                return !angular.isUndefined(_.find(this, { archived: true }));
+            }
+        }
+    });
+});
+```
+
```

Comment:
For consistency with other snippets, I suggest adding something like:

Example:

```javascript
model('Posts').all().then(function(posts) {
  if (posts.$hasArchived()) {
    // Some of the posts in the collection are archived
  }
});
```